version: '3.8'

services:
  appseed-app:
    build: .
    container_name: appseed_app
    restart: on-failure
    environment:
      - DEBUG=True
      - SERVER=.docker.localhost
      - ASSETS_ROOT=/static/assets
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.shiny.tls=true"
      - "traefik.http.routers.shiny.rule=Host(`demo.docker.localhost`)"
      - "estatcamp.url=https://demo.docker.localhost"
    networks:
      - proxy

  reverse-proxy:
    image: traefik:v2.3
    security_opt:
      - no-new-privileges:true
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ${HOME}/.certs:/etc/certs:ro
      - ${HOME}/.traefik/traefik.yml:/etc/traefik/traefik.yml:ro
      - ${HOME}/.traefik/config.yml:/etc/traefik/config.yml:ro
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.traefik=true"
    networks:
      - proxy

networks:
  proxy:
    external: true
